extension/src/test/Xliff/XliffCache.test.ts

```diff
@@ -0,0 +1,65 @@
+import { xliffCache } from "../../Xliff/XLIFFCache";
+import * as path from "path";
+import { Xliff } from "../../Xliff/XLIFFDocument";
+import * as assert from "assert";
+
+suite("XliffCache Tests", () => {
+  const cachedFilePath = path.join(
+    __dirname,
+    "../../../",
+    "src/test/resources/NAB_AL_Tools.sv-SE.xlf"
+  );
+  test("xliffCache.add()", function () {
+    xliffCache.update(
+      cachedFilePath,
+      Xliff.fromFileSync(cachedFilePath).toString()
+    );
+    assert.ok(
+      xliffCache.isCached(path.basename(cachedFilePath)),
+      "Expected file to be cached."
+    );
+    assert.ok(xliffCache.isEnabled, "Expected XliffCache to be enabled.");
+  });
+
+  test("xliffCache.isCached()", function () {
```

Comment:
This and below tests all depend on that the first test runs fine. It could be nice to add that dependency as a comment. Then we do not need to spend time of dependent errors sometime in the future.

Reply:
Or add a xliffCache.clear() in the beginning of all tests and start fresh :)

Reply:
I'll add a comment since all these test are supposed to be dependent. We don't __need__ to test them that way but it confirms that the behavior of the cache is working as intended. I'm open to changing it to independent tests if you insist on it :)

Reply:
Maybe [mocha-steps](https://www.npmjs.com/package/mocha-steps) could be an improvement to differentiate dependent/sequential tests from "regular" tests. I don't now how well they mix with suite/tests as we run it now if that goes well we could give `describe`/`it`a go as well. Which could be part of the solution for the previous given/when/then discussion.

Reply:
mocha-steps looks lika it could solve this :)

Reply:
Made a small effort to install it but quickly ran into issues with how we are running Mocha at the moment. This will be an adventure for another day :)

Reply:
Yes, skip this for now